app.js

```diff
@@ -1,4 +1,58 @@
 //  in this file create an express application - use the middle-ware built into express
+var express = require('express');
+var trucks = require('./trucks');
+var idManager = require('./idManager');
+
+idManager.setIds(trucks.getTrucks());
+
+var bodyParser = require('body-parser');
+var app = express();
+
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
+
+var serveStatic = express.static('public');
+app.use(serveStatic);
+
+app.get('/trucks', function (request, response){
+        var trucklist = trucks.getTrucks();
+        response.json(trucklist);
+});
+
+app.get('/trucks/:id', function (request, response) {
+        var id = request.params.id;
+  
+        var truck = trucks.getTruck(id);
+        if (truck) {
+                response.json(truck);
+        } else {
+                response.status(404).json('Truck not found');
+        }
+});
+
+app.post('/Trucks', function (request, response){
+        var newTruck = request.body;
+  if (newTruck) {
+      newTruck.read = false;
+      newTruck._id = idManager.getId();
+      Trucks.addTruck(newTruck);
+      response.status(201).json(newTruck);
+  } else {
+          response.status(400).json('There was a problem adding a new truck.');
+  }
+});
+
+app.delete('/trucks/:id', function (request, response) {
```

Comment:
Assignment for Module_4b does not need the DELETE route implemented (that's for Module_4b.)